src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,416 @@
+        this.getCssValue(this.element, 'bottom'),
+        this.getCssValue(this.element, 'width'));
+      this.padding = this.getCssValue(this.element, 'padding');
+      this.stickyRegionHeight = this.getCssNumber(this.stickyParent, 'height');
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this._onScrollSubscription) {
+      this._onScrollSubscription.unsubscribe();
+    }
+
+    if (this._onResizeSubscription) {
+      this._onResizeSubscription.unsubscribe();
+    }
+
+    if (this._onTouchSubscription) {
+      this._onTouchSubscription.unsubscribe();
+    }
+  }
+
+  /**
+   * getSupportList() is used to get a list of string which can be set to
+   * sticky-header's style.position and make Sticky positioning work.
+   * It returns a list of string.
+   *
+   * According to the "Position:sticky Browser compatibility" in
+   * "https://developer.mozilla.org/en-US/docs/Web/CSS/position".
+   *
+   * For Desktop: Sticky positioning works well on Chrome, Edge, Firefox and Opera. And can
+   * also work well on Safari with a "-webkit-" prefix. It only does not work on IE.
+   *
+   * For Mobile: Sticky positioning works well on Android Webview, Chrome for Android, Edge,
+   * Firefox Mobile, Opera Mobile. And can also work well on Safari Mobile with a "-webkit-" prefix.
+   * It won't always work on IE phone.
+   *
+   * The implementation references the compatibility checking in Modernizer
+   * (https://github.com/Modernizr/Modernizr/blob/master/feature-detects/css/positionsticky.js).
+   */
+  getSupportList(): Array<string> {
+    let prefixTestList = ['', '-webkit-', '-ms-', '-moz-', '-o-'];
+    let supportList: Array<string> = new Array<string>();
+    let stickyText = '';
+    for (let i = 0; i < prefixTestList.length; i++ ) {
+      stickyText += 'position:' + prefixTestList[i] + 'sticky;';
+      // Create a DOM to check if the browser support current prefix for sticky-position.
+      let div = document.createElement('div');
+      let body = document.body;
+      div.style.cssText = 'display:none;' + stickyText;
+      body.appendChild(div);
+      let isSupport = /sticky/i.test(window.getComputedStyle(div).position);
+      body.removeChild(div);
+      div = null;
+      if(isSupport == true) {
+        supportList.push(prefixTestList[i]);
+      }
+    }
+    return supportList;
+  }
+
+  /**
+   * Get the first element from this._supportList. Set it as a prefix of
+   * sticky positioning.
+   *
+   * If the this._supportList is empty, which means the browser does not support
+   * sticky positioning. Set isStickyPositionSupported as 'true' and use the original
+   * implementation of sticky-header.
+   */
+  setStrategyAccordingToCompatibility(): void {
+    let supportList = this.getSupportList();
+    if(supportList.length == 0) {
+      this.isStickyPositionSupported = true;
+    }else {
+      let prefix: string = supportList[0];
```

Comment:
Yes. Because `supportList` contains all the prefix that can make sticky positioning work in the current browser. We only need to get one prefix and make `position: prefix + 'sticky'`, then sticky position will work.